lib/OpenQA/Worker/Common.pm

```diff
@@ -336,31 +340,35 @@ sub call_websocket {
         $ua_url => {'Sec-WebSocket-Extensions' => 'permessage-deflate'} => sub {
             my ($ua, $tx) = @_;
             if ($tx->is_websocket) {
-                # keep websocket connection busy
-                $tx->send({json => {type => 'ok'}});    # Send keepalive immediately
-                $hosts->{$host}{timers}{keepalive}
-                  = add_timer("keepalive-$host", 10, sub { $tx->send({json => {type => 'ok'}}); });
-
-                $hosts->{$host}{timers}{status} = add_timer(
-                    "workerstatus-$host",
-                    15,
-                    sub {
-                        log_debug("Sending worker status to $host");
-                        $tx->send(
-                            {
-                                json => {
-                                    type => 'worker_status',
-                                    (status => 'working', job => $job) x !!($job),
-                                    (status => 'free') x !!(!$job),
-                                }});
-                    });
+
+                my $send_status = sub {
+                    log_debug("Sending worker status to $host");
+                    $tx->send(
+                        {
+                            json => {
+                                type => 'worker_status',
+                                (status => 'working', job => $job) x !!($job),
+                                (status => 'free') x !!(!$job),
+                            }});
+                };
+
+                my $i = looks_like_number($instance) ? $instance : 1;
+            # Feature scaling - we do not know the instance maximum per worker, we take DEAD_DETECTION_THRESHOLD as max.
+                my $imax = $i > DEAD_DETECTION_THRESHOLD ? $i : DEAD_DETECTION_THRESHOLD;
+                my $worker_status_timer
+                  = int(MIN_STATUS_TIMER + ((($i - 1) * (DEAD_DETECTION_THRESHOLD - MIN_STATUS_TIMER)) / ($imax - 1)));
```

Comment:
IMHO line breaks like these should be avoided so … use a temporary variable assignment?